module/input/NatNet/src/NatNet.cpp

```diff
@@ -164,23 +167,33 @@ namespace module::input {
         }
 
         // Read our metadata
-        mocap->latency      = ReadData<float>::read(ptr, version);
+        if (version < 0x03000000) {
+            mocap->latency = ReadData<float>::read(ptr, version);
+        }
         mocap->timecode     = ReadData<uint32_t>::read(ptr, version);
         mocap->timecode_sub = ReadData<uint32_t>::read(ptr, version);
 
-        // In version 2.9 natnet_timestamp went from a float to a double
-        if (version >= 0x02090000) {
+        // In version 2.7 natnet_timestamp went from a float to a double
+        if (version >= 0x02070000) {
             mocap->natnet_timestamp = ReadData<double>::read(ptr, version);
         }
         else {
             mocap->natnet_timestamp = ReadData<float>::read(ptr, version);
         }
 
+        // High res timestamps
+        if (version >= 0x03000000) {
+            mocap->mid_exposure_timestamp  = ReadData<float>::read(ptr, version);
+            mocap->data_recieved_timestamp = ReadData<float>::read(ptr, version);
+            mocap->transmit_timestamp      = ReadData<float>::read(ptr, version);
+        }
+
         short params                  = ReadData<short>::read(ptr, version);
         mocap->recording              = (params & 0x01) == 0x01;
         mocap->tracked_models_changed = (params & 0x01) == 0x02;
 
         // TODO(HardwareTeam): there is an eod thing here
```

Comment:
Given that there is a new ReadData just below this that's not getting stored anywhere is it that?
I don't remember what eod is referring to

Reply:
There is a uint32_t value sitting at the end of the packet which based on the printed hex packets appears to be a constant to indicate the end of the packet. I made the new ReadData for completeness but realistically if we don't have any use for the End of Data symbol then I can get rid of this?

Reply:
Sounds like it might be good to read and check if it is the expected eod symbol, and if it's not then throw an error saying that the packet wasn't read correctly.

Reply:
End of data makes sense. I totally would have put a comment and ignored it when I wrote it the first time